packages/devtools_app/assets/scripts/inspector_polyfill_script.dart

```diff
@@ -0,0 +1,244 @@
+              }
+              return additionalJson;
+            }),
+      );
+    }
+    return Future.value(<String, Object>{
+      'result': result,
+    });
+  }
+
+  Future<Map<String, dynamic>> setFlexFit(Map<String, String> parameters) {
+    final String id = parameters['id'];
+    final FlexFit flexFit =
+        toEnumEntry<FlexFit>(FlexFit.values, parameters['flexFit']);
+    final dynamic object = WidgetInspectorService.instance.toObject(id);
+    if (object == null) return null;
+    final render = object.renderObject;
+    final parentData = render.parentData;
+    bool succeed = false;
+    if (parentData is FlexParentData) {
+      parentData.fit = flexFit;
+      render.markNeedsLayout();
+      succeed = true;
+    }
+    return Future.value(<String, Object>{
+      'result': succeed,
+    });
+  }
+
+  Future<Map<String, dynamic>> setFlexFactor(Map<String, String> parameters) {
+    final String id = parameters['id'];
+    final String flexFactor = parameters['flexFactor'];
+    final int factor = flexFactor == 'null' ? null : int.parse(flexFactor);
+    final dynamic object = WidgetInspectorService.instance.toObject(id);
+    if (object == null) return null;
+    final render = object.renderObject;
+    final parentData = render.parentData;
+    bool succeed = false;
+    if (parentData is FlexParentData) {
+      parentData.flex = factor;
+      render.markNeedsLayout();
+      succeed = true;
+    }
+    return Future.value({'result': succeed});
+  }
+
+  Future<Map<String, dynamic>> setFlexProperties(
+      Map<String, String> parameters) {
+    final String id = parameters['id'];
+    final MainAxisAlignment mainAxisAlignment = toEnumEntry<MainAxisAlignment>(
+      MainAxisAlignment.values,
+      parameters['mainAxisAlignment'],
+    );
+    final CrossAxisAlignment crossAxisAlignment =
+        toEnumEntry<CrossAxisAlignment>(
+      CrossAxisAlignment.values,
+      parameters['crossAxisAlignment'],
+    );
+    final dynamic object = WidgetInspectorService.instance.toObject(id);
+    if (object == null) return null;
+    final render = object.renderObject;
+    bool succeed = false;
+    if (render is RenderFlex) {
+      render.mainAxisAlignment = mainAxisAlignment;
+      render.crossAxisAlignment = crossAxisAlignment;
+      render.markNeedsLayout();
+      render.markNeedsPaint();
+      succeed = true;
+    }
+    return Future.value(<String, Object>{'result': succeed});
+  }
+
+  void registerHelper(String name, ServiceExtensionCallback callback) {
+    try {
+      WidgetInspectorService.instance.registerServiceExtension(
+        name: name,
+        callback: callback,
+      );
+    } catch (e) {
+      // It is not a fatal error if some of the extensions fail to register
+      // as could be the case if some are already defined directly within
+      // package:flutter for the version of package:flutter being used.
+      print('Warning: unable to register service extension \'$name\'');
```

Comment:
can you use the DevTools logger here instead of print?

Reply:
you can't use it directly as this code is executed on the users device. However I have now plumbed it through so errors are tracked and logged back in DevTools. Did a little bit of other refactoring to make it happen.